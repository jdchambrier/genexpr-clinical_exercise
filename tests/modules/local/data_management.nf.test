nextflow_process {

    name "Test Process DATA_MANAGEMENT"
    script "modules/local/data_management.nf"
    process "DATA_MANAGEMENT"

    test("Should run without failures") {
        
        when {
            process {
                """
                // define inputs of the process here. Example:
                input[0] = file("${projectDir}/test_data/test_data.csv")
                input[1] = file("${projectDir}/bin/management.Rmd")
                container = 'rpreprocess:0.1'
                """
            }
        }

        then {
            assert process.success

            with(process.out){
                assert clean_data.size() == 1
                assert snapshot(clean_data).match("clean_data")
                assert data_management_report.size() == 1
                def lines = file(data_management_report[0]).readLines()
                assert snapshot(lines[0..10]).match("management_report_10lines")
                assert versions.size() == 1
                assert snapshot(versions).match("versions")
            }
            
        }

    }

}
