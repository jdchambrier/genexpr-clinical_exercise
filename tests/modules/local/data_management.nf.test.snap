{
    "versions": {
        "content": [
            [
                "versions.yml:md5,b77315097b93b21016accec71072609a"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "23.10.1"
        },
        "timestamp": "2025-07-26T16:42:03.814678"
    },
    "clean_data": {
        "content": [
            [
                "clean_clinical.csv:md5,6c5171a62cf177a7701abcda8d8e01a4"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "23.10.1"
        },
        "timestamp": "2025-07-26T16:25:17.268859"
    },
    "management_report_10lines": {
        "content": [
            [
                "<!DOCTYPE html>",
                "",
                "<html>",
                "",
                "<head>",
                "",
                "<meta charset=\"utf-8\" />",
                "<meta name=\"generator\" content=\"pandoc\" />",
                "<meta http-equiv=\"X-UA-Compatible\" content=\"IE=EDGE\" />",
                "",
                ""
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "23.10.1"
        },
        "timestamp": "2025-07-26T16:36:55.66173"
    }
}